pkgs/development/tools/misc/inotify-tools/default.nix

```diff
@@ -11,6 +11,10 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-EYWVSgwoMjAlc/V5kv+2jfxEqWVW/lEoIxVd+ctEMsk=";
   };
 
+  configureFlags = [
+    (lib.enableFeature fanotifySupport "fanotify")
```

Comment:
I don't see any reason why not to enable by default.
I didn't enable it by default because I didn't want to change the default behaviour.
As it depends on a kernel feature (CONFIG_FANOTIFY, which is available since kernel 5.9 according to [1](https://github.com/inotify-tools/inotify-tools/pull/134)), it might be better to condition it to this feature being activated in the kernel config. For the record, it is enabled in the NixOS kernel pkgs.linuxKernel.packages.linux_latest_libre default configuration which I am using.

Reply:
I just checked and it is seems to be actually enabled in the default common kernel configuration: [https://github.com/NixOS/nixpkgs/blob/e6db435973160591fe7348876a5567c729495175/pkgs/os-specific/linux/kernel/common-config.nix#L319](https://github.com/NixOS/nixpkgs/blob/e6db435973160591fe7348876a5567c729495175/pkgs/os-specific/linux/kernel/common-config.nix#L319)

Reply:
What do you think about enabling fanotifySupport by default and:
```suggestion
    (lib.enableFeature (fanotifySupport && linux.kernelAtLeast "5.9") "fanotify")
```
I couldn't find a way to check the specific "CONFIG_FANOTIFY" status: 
```
$ zgrep FANOTIFY /proc/config.gz
CONFIG_FANOTIFY=y
CONFIG_FANOTIFY_ACCESS_PERMISSIONS=y
```
but `linux.config.isEnabled "FANOTIFY"` yields a `false` value.

Reply:
> For the record, it is enabled in the NixOS kernel pkgs.linuxKernel.packages.linux_latest_libre default configuration which I am using.

This would be depended on the normal kernel since libre and hardened are provided on a best effort basis and where in the past just marked broken because they where badly maintained.



> I just checked and it is seems to be actually enabled in the default common kernel configuration:

Then from my side we can just activate this. We still have the 4_14 ad 5_4 kernel https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/linux-kernels.nix though. Not sure if it is realistic that someone is using them.

> What do you think about enabling fanotifySupport by default and:

This would likely always return true since linux points to the default kernel. Also integrating nixos options into packages directly is not possible.

Reply:
Ok, I just sent a new version with the option enabled by default.
I guess it is still better to present it as an option which can be deactivated using pkg.override function.